server/api_server/sync_endpoint.py

```
@@ -47,40 +47,102 @@ def handle_sync_get():
 
+    mylog("verbose", [f"[SYNC API] file_path_new={file_path_new}"])
+
+    # ---- WRITE FILE (final critical point)
     try:
+        if not isinstance(data, str):
+            data = str(data)
+
         with open(file_path_new, "w") as f:
             f.write(data)
+
     except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)